Launchpad X

worikgh opened the pull request:
Launchpad X 

The sysex message for programmer mode from page 7 Launchpad X - Programmers Reference Manual.pdf

worikgh committed:
Launchpad X

jpetazzo commented on launchpad.py:
```diff
@@ -108,6 +108,28 @@ class LaunchpadPro(LaunchPad):
+    setup = [
+        # This SysEx message switches the LaunchPad Pro to "programmer" mode
+        mido.Message("sysex", data=[0, 32, 41, 2, 12, 14, 1]),
+        # And this one sets the front/side LED
```
Does the LP X also have a "front" LED? (If not, I'll remove that sysex since it might be confusing :sweat_smile: )

worikgh replied:
I wondered about that.
I cannot see one.
I have no other Launchpad to compare it to

worikgh committed:
Initial commit

worikgh committed:
Initial commit

worikgh committed:
Be quiet!

worikgh committed:
Launchpad X

worikgh committed:
Update midi

worikgh committed:
Starts with a @reboot griode.sh entry

worikgh committed:
Prior to reformatting the SD card.  A backup.

worikgh committed:
Initial commit

worikgh committed:
Reads commands from a pipe

worikgh committed:
Commands to griode via named pipe taking shape

worikgh committed:
Changing scale by command redraws the surface of the Launchpad

worikgh committed:
Comments

worikgh committed:
Comments

worikgh committed:
Implemented command `draw`

worikgh committed:
Comments

worikgh committed:
Added command to set instrument

worikgh committed:
Initial commit

worikgh committed:
Initial commit

worikgh committed:
Upgraded?

worikgh committed:
Initial commit

worikgh committed:
Initial commit

worikgh committed:
Debugged commands

worikgh committed:
Write pid file

worikgh committed:
Edited

worikgh committed:
Enable the command pipe

worikgh committed:
Here it is....

worikgh committed:
Rename front-end

worikgh committed:
LV2 Plugins started...

worikgh committed:
Initial commit

worikgh committed:
Initial commit

worikgh committed:
Updated

worikgh committed:
Name change

worikgh committed:
Compiles

worikgh committed:
Initial commit

worikgh committed:
Commit to safety!

worikgh committed:
Improve class detection

worikgh committed:
Partially ready

worikgh committed:
Update `README.md`s to track design changes

worikgh committed:
Delete old and no longer used code

worikgh committed:
Add jack and griode commands

worikgh committed:
Updated path

worikgh committed:
Got griode commands working

worikgh committed:
Commenting

worikgh committed:
Order ports numerically

worikgh committed:
Improved Port data

worikgh committed:
Bug fix

worikgh committed:
Grouping commands

worikgh committed:
Directly command mod-host

worikgh committed:
Added griode commands

worikgh committed:
Updated writing history

worikgh committed:
Reads commands off stdin

worikgh committed:
Fix history writing

worikgh committed:
Tidied up

worikgh committed:
Add control for root note

worikgh committed:
Updated

worikgh committed:
Initial commit

worikgh committed:
Minor fixups

worikgh committed:
iIt is what it is....

worikgh committed:
No longer uses a pipe to send commands to `griode`

worikgh committed:
Set default root note.  Not persistant

worikgh committed:
Major changes and development

worikgh committed:
Evolving....

worikgh committed:
Set root note via offset from AQ

worikgh committed:
Initial commit

worikgh committed:
Renamed

worikgh committed:
Initial commit

worikgh committed:
Times in log messages and selcted instrument in Model

worikgh committed:
Run commands to set up the instrument

worikgh committed:
Minor update

worikgh committed:
Compile chrono

worikgh committed:
Fix documentaton

worikgh committed:
Visual indication of which instrument selected

worikgh committed:
Implement command "root" properly

worikgh committed:
Started illustrating setting up the instrument

worikgh committed:
Quieten timing warnings so they are only emitted at debug level

worikgh committed:
Set timing information into log method.  Set font onfront end

worikgh committed:
Give the realnames for instruments so the front end can see them

worikgh committed:
Animate selection while being processed

worikgh committed:
Fix(!) home directory

Home directory is hard coded.  Bad, bad, bad!

worikgh committed:
minor edits

worikgh committed:
Get LEDs to display correctly if key shifted by a root command

worikgh committed:
Minor change

worikgh committed:
Translate jack ports

worikgh committed:
Translate jack ports

worikgh committed:
Initial commit

worikgh committed:
Improved error handling and uses `find_lv2`

worikgh committed:
A mostly working build

worikgh committed:
Explicit name spacing for ws types.

For clarity

worikgh committed:
Compiles

Made big changes.  Principally `ServerState`

worikgh committed:
Minor change

worikgh committed:
Still a chaotic mess!

worikgh committed:
Two examples of songs/settings

worikgh committed:
Compiles

worikgh committed:
Revove dead code

worikgh committed:
Minor edit

worikgh committed:
Connect up the song files and internal data structures

worikgh committed:
Minor edits

worikgh committed:
Compiles cleanly

worikgh committed:
Minor edits

worikgh committed:
Reorder functions to group more rationally

worikgh committed:
minor edit

worikgh committed:
Process INIT from server

worikgh committed:
Cleaned up a bit

worikgh committed:
Cleaning up....

worikgh committed:
Starting to read from pedal

worikgh committed:
Reads keys for pedal

worikgh committed:
Missed code for ncurses

worikgh committed:
Reviewed and updated niitialisation code for `ServerState`

worikgh committed:
Added ncurses

worikgh committed:
Initial commit

I think....

worikgh committed:
New environment variable to find directory and start of pedlal code

worikgh committed:
Reorganised

worikgh committed:
Add Debug to structures

worikgh committed:
Adding Pedal

worikgh committed:
Edited for clarity

worikgh committed:
Inintial commit

worikgh committed:
Minor edits

worikgh committed:
Use env var PATH_MI_ROOT

worikgh committed:
Use env var PATH_MI_ROOT

worikgh committed:
Typo

worikgh committed:
Added pedals

worikgh committed:
Got rid of unneeded abstractions

worikgh committed:
Do not need these abstractions

worikgh committed:
Add persistent state

worikgh committed:
Implemented saving state

worikgh committed:
Initial commit.

Copied from mod-devices github site

worikgh committed:
Started implementation of pedal switching

worikgh committed:
Updated logging so it goes to a file

worikgh committed:
Updated names and comments

worikgh committed:
Tidy up...

worikgh committed:
Working!

worikgh committed:
Make valid

worikgh committed:
Better logging

worikgh committed:
typo fix

worikgh committed:
Minor changes

worikgh committed:
DO not start front end (yet)

worikgh committed:
Fixup case error

worikgh committed:
Move client to its own directory

It was not happy that the server uses `ncurses-rs` but the server is entirely different so they needed to be separated

worikgh committed:
Move client to its own directory

It was not happy that the server uses `ncurses-rs` but the server is entirely different so they needed to be separated

worikgh committed:
Move client to its own directory

It was not happy that the server uses `ncurses-rs` but the server is entirely different so they needed to be separated

worikgh committed:
Move client to its own directory

It was not happy that the server uses `ncurses-rs` but the server is entirely different so they needed to be separated

worikgh committed:
Move client to its own directory

It was not happy that the server uses `ncurses-rs` but the server is entirely different so they needed to be separated

worikgh committed:
Bugfix

worikgh committed:
Moving client....

worikgh committed:
Formatted (cargo fmt) and invalid instruments no longer panic

worikgh committed:
Working.

worikgh committed:
Tweek colours

worikgh committed:
Quietened.  Nothing on stdout now

worikgh committed:
Renable pedal via stdin

worikgh committed:
Redircting output from sub-processes

worikgh committed:
Pedal display

worikgh committed:
Manage to delete client communication channels when client closes

worikgh committed:
Code cleanups

worikgh committed:
Connect to VU meters if present

worikgh committed:
Update for new CSS

worikgh committed:
Minor edits

worikgh committed:
Not specific to particular pedals

worikgh committed:
Minor change

worikgh committed:
idempotent and run front end

worikgh committed:
Logging changes

worikgh committed:
Minor change

worikgh committed:
Updated

worikgh committed:
Updated

worikgh committed:
Replumbed....
